bool appear
float64 cx_left
float64 cx_right
float64 area
